packages/data-point/lib/reducer-path/factory.js

```diff
@@ -13,27 +13,78 @@ const REDUCER_PATH = (module.exports.type = 'ReducerPath')
 function ReducerPath () {
   this.type = REDUCER_PATH
   this.name = ''
+  this.body = undefined
   this.asCollection = false
 }
 
 module.exports.ReducerPath = ReducerPath
 
+/**
+ * @param {*} source
+ * @returns {boolean}
+ */
 function isPath (source) {
   return _.isString(source) && source.charAt(0) === '$'
 }
 
 module.exports.isPath = isPath
 
+const getAccValue = acc => acc.value
+
+module.exports.getAccValue = getAccValue
```

Comment:
This is definitely in the right direction of precompiled reducers!

Some notes regarding these methods:

- Add specific unit tests: can you add specific tests for all of the functions exposed, I know you have a set of other tests that use them through `body` but those become more of integration tests than actual unit tests. 

- Move to named functions: move the arrow functions to **named functions** It will deliver a better experience for our users. The reason I have favored named functions vs arrow functions is that even though they are awesome for currying they do not result in a good debugging experience - someone reading an error call-stack.

- Try using more descriptive function names, and if possible avoid shortening words. The names in their context of the file are fine, but for an outsider, they might not mean much. 

List of methods and some naming suggestions: 

- `getAccValue` >> `getAccumulatorValue`
- `getFromAcc`  >> `getAccumulatorPath`
- `getFromAccValue ` >> `getAccumulatorValuePath`
- `mapFromAccValue` >> `mapAccumulatorValuePath`

as an example: 

```js
const getFromAccValue = jsonPath => acc => _.get(acc.value, jsonPath)
```

to: 

```js
function getAccumulatorValuePathResolver(jsonPath) {
 return function getAccumulatorValuePath(acc) { 
   return _.get(acc.value, jsonPath)
 }
}
```

Reply:
not for now, but I feel we should prob create a compiler.js file where these methods live so they are not inside the factory file itself.... not or now, just something to think about

Reply:
> Add specific unit tests: can you add specific tests for all of the functions exposed, I know you have a set of other tests that use them through body but those become more of integration tests than actual unit tests.

IMO the integration tests are more valuable than the unit tests since it's that much less code to maintain over time. We still get full coverage with these tests, so unit tests would be redundant I think.

Reply:
I understand the point, but still rather have them unit tested, its been my experience when I then later decide to use X method somewhere else, then I have to replicate the 'integration' test on the other second implementation

Reply:
@acatl Just pushed all the changes you requested!